lib/Serialization/Deserialization.cpp

```diff
@@ -3732,10 +3734,10 @@ class swift::DeclDeserializer {
 
     MF.configureGenericEnvironment(extension, genericEnvID);
 
-    auto baseTy = MF.getType(baseID);
+    auto extendedType = MF.getType(extendedTypeID);
     ctx.evaluator.cacheOutput(ExtendedTypeRequest{extension},
-                              std::move(baseTy));
-    auto nominal = extension->getExtendedNominal();
+                              std::move(extendedType));
+    auto nominal = dyn_cast<NominalTypeDecl>(MF.getDecl(extendedNominalID));
```

Comment:
You need to call ctx.evaluator.cacheOutput() to cache the result of ExtendedNominalRequest as well, so that the request doesn't get re-computed.

Reply:
I don't quite understand -- in some cases, caching is automatic whereas not in other cases? Or does it always have to be done manually?